Source: https://src.fedoraproject.org/rpms/efitools/blob/f43/f/efitools-bsc1220470-fix-gcc14-errors.patch

From 1981f172cff6992dccd80d0ca240b2ebdf1664cf Mon Sep 17 00:00:00 2001
From: Gary Lin <glin@suse.com>
Date: Fri, 8 Mar 2024 11:35:39 +0800
Subject: [PATCH] Define _XOPEN_SOURCE for strptime()

Per manpage, strptime() requires _XOPEN_SOURCE being defined and gcc 14
would issue an error for that.

Define _XOPEN_SOURCE to avoid the error.

Signed-off-by: Gary Lin <glin@suse.com>
---
 flash-var.c         | 1 +
 sign-efi-sig-list.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/flash-var.c b/flash-var.c
index aa10ae6..d5f5f2f 100644
--- a/flash-var.c
+++ b/flash-var.c
@@ -1,3 +1,4 @@
+#define _XOPEN_SOURCE
 #include <stdlib.h>
 #include <stdint.h>
 #include <sys/types.h>
diff --git a/sign-efi-sig-list.c b/sign-efi-sig-list.c
index 94bd7d4..c68c78b 100644
--- a/sign-efi-sig-list.c
+++ b/sign-efi-sig-list.c
@@ -5,6 +5,7 @@
  */
 #include <stdint.h>
 #define __STDC_VERSION__ 199901L
+#define _XOPEN_SOURCE
 #include <efi.h>
 #ifdef CONFIG_arm
 /* FIXME:
-- 
2.35.3

